ipc/src/server.rs

```diff
@@ -187,6 +189,7 @@ impl<M: Metadata, S: Middleware<M>> ServerBuilder<M, S> {
 
 				Ok(())
 			});
+			start_signal.send(Ok(())).expect("Cannot fail since receiver never dropped before receiving");
```

Comment:
`start_signal` was triggered before the server was fully set up. After fixing the order of initialisation, we no longer need `thread::sleep` in tests